src/Diagnostics.DataProviders/AscClient.cs

```diff
@@ -215,6 +231,10 @@ private static T CastTo<T>(object obj)
 
         private async Task<T> GetAscResponse<T>(HttpRequestMessage requestMessage, bool isBlobRequest, CancellationToken cancellationToken = default(CancellationToken))
         {
+            if (!string.IsNullOrWhiteSpace(SubscriptionLocationPlacementId) && SubscriptionLocationPlacementId.Equals("geos_2020-01-01", StringComparison.CurrentCultureIgnoreCase))
+            {
+                throw new HttpRequestException(MethodNotAllowed);
```

Comment:
Should we throw InvalidOperationException here?

https://docs.microsoft.com/en-us/dotnet/api/system.invalidoperationexception?redirectedfrom=MSDN&view=netframework-4.8